# Table of Contents
## 01. Importing Libraries
## 02. Importing Data
## 03. Checking Data
## 04. Exporting Data

# 1. Importing Libraries

In [2]:
#import libaries
import pandas as pd
import numpy as np
import os

# 2. Import Data

In [3]:
path = r'/Users/lianabulte/Career Foundry/2023 02 Instacart Basket Analysis'

In [4]:
df_opc = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))

In [5]:
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_checked.csv'), index_col = False)

# 3. Checking Data

In [6]:
# check shape of df_opc
df_opc.shape

(30356421, 10)

In [7]:
# check summary of df_opc
df_opc.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2398795        1             2                   3                  7   
1   2398795        1             2                   3                  7   
2   2398795        1             2                   3                  7   
3   2398795        1             2                   3                  7   
4   2398795        1             2                   3                  7   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  
0                    15.0         196                  1          1   both  
1                    15.0       10258                  2          0   both  
2                    15.0       12427                  3          1   both  
3                    15.0       13176                  4          0   both  
4                    15.0       26088                  5          1   both

In [8]:
#drop merge flag from df_opc
df_opc = df_opc.drop(columns = ['_merge'])

In [9]:
df_opc.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2398795        1             2                   3                  7   
1   2398795        1             2                   3                  7   
2   2398795        1             2                   3                  7   
3   2398795        1             2                   3                  7   
4   2398795        1             2                   3                  7   

   days_since_prior_order  product_id  add_to_cart_order  reordered  
0                    15.0         196                  1          1  
1                    15.0       10258                  2          0  
2                    15.0       12427                  3          1  
3                    15.0       13176                  4          0  
4                    15.0       26088                  5          1

In [10]:
# check shape of df_prods
df_prods.shape

(49672, 6)

In [11]:
#check summary of df_prods
df_prods.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [12]:
# drop the extra index and reset index
df_prods = df_prods.reset_index(drop = True)
df_prods = df_prods.drop(columns = 'Unnamed: 0')

In [13]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [14]:
df_prods.shape

(49672, 5)

#### Confirm suitable way to combine dataframe to products data set

In [15]:
# merge the two dataframes
df_opc_merged = df_opc.merge(df_prods, on = 'product_id', indicator = True)

In [17]:
df_opc_merged.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2398795        1             2                   3                  7   
1    473747        1             3                   3                 12   
2   2254736        1             4                   4                  7   
3    431534        1             5                   4                 15   
4   3367565        1             6                   2                  7   

   days_since_prior_order  product_id  add_to_cart_order  reordered  \
0                    15.0         196                  1          1   
1                    21.0         196                  1          1   
2                    29.0         196                  1          1   
3                    28.0         196                  1          1   
4                    19.0         196                  1          1   

  product_name  aisle_id  department_id  prices _merge  
0         Soda        77              7     9.0   both  
1         Soda        77              7     9.0   both  
2         Soda        77              7     9.0   both  
3         Soda        77              7     9.0   both  
4         Soda        77              7     9.0   both

#### Confirm results of the merge flag

In [18]:
df_opc_merged['_merge'].value_counts()

both          30328763
left_only            0
right_only           0
Name: _merge, dtype: int64

In [19]:
df_opc_merged.shape

(30328763, 14)

the merge was not a full match as the totals rows did not match the intial rows of df_opc (30356421)

# Exporting Data

In [20]:
# export data to pkl 
df_opc_merged.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_merged.pkl'))